src/Microsoft.DotNet.ProjectModel/DependencyContextBuilder.cs

```diff
@@ -68,8 +68,12 @@ private static CompilationOptions GetCompilationOptions(CommonCompilerOptions co
 
             var libraryAssets = runtime ? export.RuntimeAssemblies : export.CompilationAssemblies;
 
-            foreach (var libraryDependency in export.Library.Dependencies)
+            foreach (var libraryDependenciesGroup in export.Library.Dependencies.GroupBy(d => d.Name))
             {
+                LibraryRange libraryDependency = libraryDependenciesGroup
+                    .OrderByDescending(d => d.Target == LibraryType.ReferenceAssembly)
```

Comment:
There may be multiple dependencies with the same name and we want to prefer `ReferenceAssembly` but if it doesn't exist take whats left